src/services/fimsService.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * This service interacts with the IO FIMS API
+ */
+
+import {
+  IResponseErrorConflict,
+  IResponseErrorInternal,
+  IResponseErrorValidation,
+  IResponseSuccessAccepted,
+  IResponseSuccessJson,
+  ResponseErrorConflict,
+  ResponseErrorInternal,
+  ResponseErrorValidation,
+  ResponseSuccessAccepted,
+  ResponseSuccessJson,
+} from "@pagopa/ts-commons/lib/responses";
+
+import { EmailString, FiscalCode } from "@pagopa/ts-commons/lib/strings";
+
+import { AccessHistoryPage } from "generated/io-fims-api/AccessHistoryPage";
+import { ExportRequest } from "generated/io-fims-api/ExportRequest";
+
+import { IoFimsAPIClient } from "../clients/io-fims";
+
+import {
+  ResponseErrorStatusNotDefinedInSpec,
+  withCatchAsInternalError,
+  withValidatedOrInternalError,
+} from "../utils/responses";
+
+const invalidRequest = "Invalid request";
+
+export default class FimsService {
+  constructor(private readonly ioFimsApiClient: ReturnType<IoFimsAPIClient>) {}
+
+  public readonly getAccessHistory = (
+    fiscalCode: FiscalCode,
+    page?: string
+  ): Promise<
+    | IResponseErrorInternal
+    | IResponseErrorValidation
+    | IResponseSuccessJson<AccessHistoryPage>
+  > =>
+    withCatchAsInternalError(async () => {
+      const validated = await this.ioFimsApiClient.getAccessHistory({
+        page,
+        user: fiscalCode,
+      });
+      return withValidatedOrInternalError(validated, (response) => {
+        switch (response.status) {
+          case 200:
+            return ResponseSuccessJson(response.value);
+          case 422:
+            return ResponseErrorValidation(
```

Comment:
Ditto, this will return `400`

Reply:
`fims-user-func` returns `422` so we have to check for this status here (but `io-backend` will map `422` to `400`)

Reply:
Yess, I meant that `ResponseErrorValidation` returns `400`, anyway if you updated the OpenAPI it's fine to me 😄